controllers/snapshot/snapshot_controller.go

```diff
@@ -96,10 +97,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 	logger = logger.WithApp(*application)
 
-	component, err := loader.GetComponentFromSnapshot(r.Client, ctx, snapshot)
+	var component *applicationapiv1alpha1.Component
+	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+		component, err = loader.GetComponentFromSnapshot(r.Client, ctx, snapshot)
+		if err != nil {
+			logger.Error(err, "Failed to get Component from the Snapshot")
+			return err
+		}
+		return nil
+	})
 	if err != nil {
-		logger.Error(err, "Failed to get Component from the Snapshot")
-		return ctrl.Result{}, err
+		logger.Error(err, "Could not get component from Snapshot.  Component may have been removed.  Declining to proceed with reconciliation")
+		return ctrl.Result{}, nil
```

Comment:
Here you should exactly check if error is NotFound error, and only then stop reconciliation.
```
errors.IsNotFound(err)
```